utils/dependency_management.py

```diff
@@ -294,10 +294,17 @@ def check_requirements(extras, github_annotate):
 
         for requirement_abstract in requirements_abstract:
             for requirement_concrete in requirements_concrete:
-                version = Specifier(str(requirement_concrete.specifier)).version
-                if canonicalize_name(requirement_abstract.name) == canonicalize_name(
-                    requirement_concrete.name
-                ) and requirement_abstract.specifier.contains(version):
+                if '@' in str(requirement_concrete):
+                    version = str(requirement_concrete).split('@')[1]
+                    abstract_contains = version in str(requirement_abstract)
+                else:
+                    version = Specifier(str(requirement_concrete.specifier)).version
+                    abstract_contains = requirement_abstract.specifier.contains(version)
+
+                if (
+                    canonicalize_name(requirement_abstract.name) == canonicalize_name(requirement_concrete.name)
+                    and abstract_contains
+                ):
```

Comment:
I plan to keep it, as it's very useful to pass CI when we make PRs like this, that are hooked to another PR, or branch of other repo with `@`

The problem is `@` is not listed as a valid `specifier` in `class Specifier`.
This little change, basically, accepts `@` as a valid specifier and will check if a hooked dependency is to the same "version" across all files, `requirement-xx` and `enviroment.yml` , etc... 

This way, apart of this nice check, the dependency test fails and it still triggers the main unit tests `test-presto` , `test-3.xx` for such PRs.. (otherwise it won't)

Reply:
Thanks!

Reply:
As we discussed, this feature is already covered in the new PR https://github.com/aiidateam/aiida-core/pull/6640. 
So I keep the changes temporarily for this PR only, and will revert `'utils/dependency_management.py'` before any merge.